witness/src/Dscp/Witness/Web/Logic.hs

```diff
@@ -94,3 +95,24 @@ getTransactionInfo =
     runSdMRead . getTxMaybe >=>
     nothingToThrow TransactionNotFound >=>
     return . toTxInfo
+
+getHashType :: WitnessWorkMode ctx m => Text -> m HashIs
+getHashType hash = fmap (fromMaybe HashIsUnknown) . runMaybeT . asum $
+    [isBlock, isAddress, isTx] <*> [hash]
+  where
+    is hashType decode getMaybe =
+        MaybeT . return . rightToMaybe . decode >=>
+        MaybeT . getMaybe >=>
+        MaybeT . return . Just . const hashType
```

Comment:
This is subjective, but here you seem to write it in the long way in purpose.
First line can be written like `Right id <- decode text`, and the third one is just `pure hashType` if written with do-notation.

Reply:
Not sure I get your idea, but this is as far as my Haskell jutsu gets without multi-storey `case`s

Reply:
Since you are in `MaybeT` anyway you could just use partial pattern-matches of do-notation which use `fail` - instead of `case`, but nvm